build.gradle.kts

```diff
@@ -104,54 +104,130 @@ extra.apply {
   val baseFilename = "${project.name}-${version}"
   set("targetFilePathBase", "${buildDir}/dist/${baseFilename}")
 
+  val shadowJarFilename = "${baseFilename}-all.jar"
+  set("shadowJarFilename", shadowJarFilename)
+
   val javaHome = System.getProperty("java.home") ?: throw GradleException("java.home is not set")
-  val jpackage = javaHome + File.separator + "bin" + File.separator + "jpackage"
+  val jpackage = "${javaHome}/bin/jpackage"
   val jPackageCmd = if (jpackage.contains(" ")) "\"" + jpackage + "\"" else jpackage
   set("jPackageCmd", jPackageCmd)
-  val sharedParameters = ArrayList<String>(listOf(
+
+  val copyrights = "Copyright ©2001–${SimpleDateFormat("yyyy").format(Date())} ${project.name} developers"
+
+  // Platform-agnostic jpackage parameters shared across all the builds.
+  val params = ArrayList<String>(listOf(
       jPackageCmd,
     "--verbose",
       "--input", "${buildDir}/libs",
       "--main-class", "com.cburch.logisim.Main",
-      "--main-jar", "${baseFilename}-all.jar",
+      "--main-jar", shadowJarFilename,
       "--app-version", version,
-      "--copyright", "\"Copyright ©2001–${SimpleDateFormat("yyyy").format(Date())} ${project.name} developers\"",
+      "--copyright", "\"${copyrights}\"", // Must be quoted for shell happines.
       "--dest", "${buildDir}/dist"
   ))
-  val linuxParameters = ArrayList<String>(listOf(
+  set("sharedParameters", params)
+
+  // Linux (DEB/RPM) specific settings for jpackage.
+  params.addAll(arrayOf(
```

Comment:
You set the sharedParameters to params which is good. Then you changed params by adding the Linux specific things. This changes the object that sharedParameters uses. The result is that sharedParameters now has Linux specific options. This causes createApp (and I assume createMsi) to fail. The original code created a new ArrayList object for the Linux things rather than adding to the old.

Reply:
I already fixed that locally. This approach was fine as we were constructing `ArrayList` to wrap params. Once I dropped that it make no sense. Just add `.clone()` to line 128:

    set("sharedParameters", params.clone())

Reply:
Yes, clone() works.